skimage/registration/_tvl1.py

```diff
@@ -0,0 +1,206 @@
+    f1 = dt / tau
+    tol *= I0.size
+
+    flow = flow0
+
+    g = np.zeros((I0.ndim, ) + I0.shape)
+    proj = np.zeros((I0.ndim, I0.ndim, ) + I0.shape)
+
+    s_g = [slice(None), ] * g.ndim
+    s_p = [slice(None), ] * proj.ndim
+    s_d = [slice(None), ] * (proj.ndim-2)
+
+    for _ in range(nwarp):
+        if prefilter:
+            flow = ndi.filters.median_filter(flow, [1]+I0.ndim*[3])
+
+        wI1 = warp(I1, grid+flow, mode='nearest')
+        grad = np.array(np.gradient(wI1))
+        NI = (grad*grad).sum(0)
+        NI[NI == 0] = 1
+
+        rho_0 = wI1 - I0 - (grad*flow0).sum(0)
+
+        for _ in range(niter):
+
+            # Data term
+
+            rho = rho_0 + (grad*flow).sum(0)
+
+            idx = abs(rho) <= f0 * NI
+
+            flow_ = flow
+
+            flow_[:, idx] -= rho[idx]*grad[:, idx]/NI[idx]
+
+            idx = ~idx
+            srho = f0 * np.sign(rho[idx])
+            flow_[:, idx] -= srho*grad[:, idx]
+
+            # Regularization term
+            flow = flow_.copy()
+
+            for idx in range(flow.shape[0]):
+                s_p[0] = idx
+                for _ in range(2):
+                    for ax in range(flow.shape[0]):
+                        s_g[0] = ax
+                        s_g[ax+1] = slice(0, -1)
+                        g[tuple(s_g)] = np.diff(flow[idx], axis=ax)
+                        s_g[ax+1] = slice(None)
+
+                    norm = np.sqrt((g ** 2).sum(0))[np.newaxis, ...]
+                    norm *= f1
+                    norm += 1.
+                    proj[idx] -= dt * g
+                    proj[idx] /= norm
+
+                    # d will be the (negative) divergence of p[idx]
+                    d = -proj[idx].sum(0)
+                    for ax in range(flow.shape[0]):
+                        s_p[1] = ax
+                        s_p[ax+2] = slice(0, -1)
+                        s_d[ax] = slice(1, None)
+                        d[tuple(s_d)] += proj[tuple(s_p)]
+                        s_p[ax+2] = slice(None)
+                        s_d[ax] = slice(None)
+
+                    flow[idx] = flow_[idx] + d
+
+        flow0 -= flow
+        if (flow0*flow0).sum() < tol:
+            break
+
+        flow0 = flow
+
+    return flow
+
+
+def tvl1(I0, I1, dt=0.2, lambda_=15, tau=0.3, nwarp=5, niter=10,
+         tol=1e-4, prefilter=False):
+    """Coarse to fine TV-L1 optical flow estimator.
+
+    TV-L1 ia popular algorithm for optical flow estimation intrudced
+    by Zack et al. [1]_, improved in [2]_ and detailed in [3]_.
+
+    Parameters
+    ----------
+    I0 : ~numpy.ndarray
+        The first gray scale image of the sequence.
+    I1 : ~numpy.ndarray
+        The second gray scale image of the sequence.
+    dt : float
+        Time step of the numerical scheme. Convergence is proved for
+        values dt < 0.125, but it can be larger for faster
+        convergence.
+    lambda_ : float
+        Attachement parameter. The smaller this parameter is,
```

Comment:
```suggestion
        Attachment parameter. The smaller this parameter is,
```

I wonder if we should name these "attachment" and "tightness", and specify in the docstring that they correspond to "lambda" and "tau" from the paper. What do you think?